mlfinlab/portfolio_optimization/risk_estimators.py

```diff
@@ -338,13 +426,16 @@ def denoise_covariance(self, cov, tn_relation, denoise_method='const_resid_eigen
         if denoise_method == 'target_shrink':
             # Based on the threshold, de-noising the correlation matrix
             corr = self._denoised_corr_targ_shrink(eigenval, eigenvec, num_facts, alpha)
+        elif denoise_method == 'spectral':
+            # Based on the threshold, de-noising the correlation matrix
+            corr = self._denoised_corr_spectral(eigenval, eigenvec, num_facts)
         else: # Default const_resid_eigen method
             # Based on the threshold, de-noising the correlation matrix
             corr = self._denoised_corr(eigenval, eigenvec, num_facts)
 
         # Detone the correlation matrix if needed
         if detone:
-            corr = self._detoned_corr(corr, market_component)
+            corr = self._detoned_corr(corr, eigenval, eigenvec, num_facts, market_component)
```

Comment:
Was this change made intentionally?

Reply:
I'll fix this one. Next time please double-check to create branched from `develop` and not `master`.

Reply:
Fixed de-toning function and unit test.